### DISCUSSION TITLE
Phrase Matcher space sensitive issue

### DISCUSSION THREAD

**Discussion by AnishaMohandass** [explosion/spaCy#4926] [Help: Coding & Implementations] [answered]
terms = ["Barack Obama", "Angela Merkel", "Washington, D.C."]
doc = nlp("German Chancellor Angela Merkel and US President Barack Obama "
          "converse in the Oval Office inside the White House in Washington, D.C.")

If I enter an extra space between the words "Barack  Obama", the phrase matcher does not work since it is space sensitive.
Is there a way to overcome this space sensitive issue?

## Your Environment

* Operating System: Windows 8
* Python Version Used: 3.7
* spaCy Version Used: 2.2.3
* Environment Information: Conda

**Comment by svlandeg** [comment, marked as answer]
The `PhraseMatcher` is not just space sensitive - it really requires the terms to appear in the text exactly as in the `terms` list. 

One option is to expand your `terms` list with lexical variants such as added spaces, but that's not a very elegant solution. 

Another option is to pre-process your input texts and remove multiple spaces, if those are a frequent problem in your input text (do this before you do any spaCy processing at all).

A final option I can think of, is to look into [Matching regular expressions on the full text](https://spacy.io/usage/rule-based-matching#regex-text). With regular expressions, you can succintly match on various lexical spelling variants.

**Comment by AnishaMohandass** [comment]
Thanks for your response @svlandeg.
Pre-processing the input text to remove multiple spaces, works fine.

input_text = "German Chancellor Angela Merkel and US President Barack     Obama converse in the Oval Office inside the White House in Washington, D.C."
sentence = re.sub(' +',' ', input_text)
